calculator.js

```diff
@@ -1,126 +1,134 @@
-function Calculator() {
-    'use strict';
-
-    let newInput = "0";
-    let oldInput = "";
-    let result = "";
-    this.input = ko.observable(newInput);
-    let operation = "";
-    let history = [];
-
-    //--------ADD NUMBER---------
-    this.addNumber = (number) => () => {
-        if (newInput === "0") {
-            newInput = number.toString();
+(function (factory) {
+    if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));
+    } else {
+        window.Calculator = factory(ko);
+    }
+} (function (ko) {

```

Comment:
This is where our "factory function" begins. Notice how it takes ko as a dependency.